src/types.rs

```diff
@@ -159,6 +195,46 @@ pub struct Block {
     pub mix_hash: Option<String>,
 }
 
+impl ToChecksummed for Block {
```

Comment:
don't need trait for it should just make the Block::from something like `Block::from_other` and make it take a bool

Reply:
Or better, can put a generic argument to it like `Fn(&[u8]) -> String` and pass encode_hex or checksumming implemtation based on if statement

Reply:
Maybe we can chat on a call? I'm not sure I understand the benefit of what you're suggesting for this scenario.

Reply:
Should never use generics/traits/macros unless it is really needed. It makes code much harder to understand. With a function call you can crtl+click and go to what it is doing for example. But with traits etc. you need to go to trait understand it check how it is used etc. etc.

This is just a general style we use in server code as well.

Generics can be useful here because it removes a branch from code but still doesn't feel worth it since it will take the same branch every time anyways and it will be predicted.

I suggest just making a function like `encode_address` or similar and make it take a bool and have an if inside. Then can pass the bool around